app/values/file_info.rb

```diff
@@ -0,0 +1,24 @@
+class FileInfo
+  attr_reader :file_info
+  delegate :[], :has_key?, to: :file_info
+  def initialize(file_info)
+    @file_info = file_info
+  end
+
+  def to_h
+    file_info
+  end
+
+  def file_name
+    file_info["file_name"]
+  end
+
+  def file_path
+    FilePath.new(file_info["url"])
+  end
+
+  def mime_type
```

Comment:
Should we allow `file_info['mime_type']` to override the extension-based mime type guessing?  I don't have any use case for doing that now, but it's always nice to be able to explicitly set the mime type when you know a more specific type than what the extension would get you.